src/Common/Auth/OneTimeAuth.php

```diff
@@ -0,0 +1,340 @@
+<?php
+
+/**
+ * OneTimeAuth Class
+ * Service for Onetime token creation, routing and auth
+ *
+ * @package   OpenEMR
+ * @link      http://www.open-emr.org
+ * @author    Jerry Padgett <sjpadgett@gmail.com>
+ * @copyright Copyright (c) 2023 Jerry Padgett <sjpadgett@gmail.com>
+ * @license   https://github.com/openemr/openemr/blob/master/LICENSE GNU General Public License 3
+ */
+
+namespace OpenEMR\Common\Auth;
+
+use DateInterval;
+use DateTime;
+use MyMailer;
+use OpenEMR\Common\Crypto\CryptoGen;
+use OpenEMR\Common\Logging\SystemLogger;
+use OpenEMR\Common\Session\SessionUtil;
+use OpenEMR\Common\Utils\RandomGenUtils;
+
+class OneTimeAuth
+{
+    private $scope;
+    private $context;
+
+    public function __construct($context = 'portal', $scope = 'redirect')
+    {
+        // scope = portal/service tasks (reset, register). context = portal, patient etc.
+        $this->scope = $scope;
+        $this->context = $context;
```

Comment:
Something that can increase performance is creating the CryptoGen instance in __construct and storing it as a private variable (this instance will cache the keys as they are used so it doesn't then need to keep collecting and unencrypting the key from drive if done more than once)

Reply:
I usually do this but have been lazy. thanks for suggestion.